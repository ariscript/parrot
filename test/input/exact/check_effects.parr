def reset(t):
    t[0] := 0

def inc(t):
    t[0] := (t[0] + 1)

check:
    val box = (0,),
    # these tests are supposed to fail, but they show evaluation order
    print(inc(box)) is print(inc(box)),
    reset(box);
    print(inc(box)) is print(inc(box)) because print(inc(box)).

nil
