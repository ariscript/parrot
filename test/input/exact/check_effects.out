1
2
failed test at 10:4-10:38: the left and right sides were not `equal` (1 and 2)
1
2
3
failed test at 11:4-12:62: the explanation and right side were not `equal` (3 and 2)
failed test at 11:4-12:62: the left and right sides were not `equal` (1 and 2)
check block: 0/2 tests passed
nil
